### PR TITLE
Update to use 0.65 syntax and latest beta.

### PR DESCRIPTION
**Pull request by cgranade**

### CODE REVIEW

File: Chemistry/SimulateHubbardHamiltonian/HubbardSimulation.qs [hunk 1]
```diff
@@ -22,7 +21,7 @@ namespace Microsoft.Quantum.Chemistry.Samples.Hubbard {
         mutable spinUpIndices = new Int[nFilling / 2];
         
         for (idx in 0 .. nFilling / 2 - 1) {
-            set spinUpIndices[idx] = idx;
+            set spinUpIndices w/= idx <- idx;
```
Comment (bettinaheim) [inline Chemistry/SimulateHubbardHamiltonian/HubbardSimulation.qs]: this also looks like a case where an initializer function from the array module should be used...

File: Samples/src/H2SimulationCmdLine/Operation.qs [hunk 2]
```diff
@@ -30,61 +30,61 @@ namespace Microsoft.Quantum.Samples.H2Simulation {
...
-        set bondCoefficients[46] = [0.1098, 0.0549, 0.3521, 0.1412, 0.1412];
-        set bondCoefficients[47] = [0.1078, 0.0572, 0.3491, 0.1422, 0.1422];
-        set bondCoefficients[48] = [0.1059, 0.0594, 0.3461, 0.1432, 0.1432];
-        set bondCoefficients[49] = [0.1042, 0.0614, 0.3433, 0.1441, 0.1441];
-        set bondCoefficients[50] = [0.1026, 0.0632, 0.3406, 0.145, 0.145];
-        set bondCoefficients[51] = [0.1011, 0.0649, 0.3379, 0.1458, 0.1458];
-        set bondCoefficients[52] = [0.0997, 0.0665, 0.3354, 0.1467, 0.1467];
-        set bondCoefficients[53] = [0.0984, 0.0679, 0.3329, 0.1475, 0.1475];
+        let bondCoefficients = new Double[][nBondLengths]
+            w/ 0 <- [0.5678, -1.4508, 0.6799, 0.0791, 0.0791]
```
Comment (bettinaheim) [inline Samples/src/H2SimulationCmdLine/Operation.qs]: Why not just use an array-of-array literal here(for the entire bondCoefficients )? The use of mutability seems rather unnecessary.

File: Samples/src/H2SimulationGUI/H2SimulationOperations.qs [hunk 3]
```diff
@@ -27,64 +27,65 @@ namespace Microsoft.Quantum.Samples.H2Simulation {
...
-        set bondCoefficients[45] = [0.1119, 0.0523, 0.3553, 0.1402, 0.1402];
-        set bondCoefficients[46] = [0.1098, 0.0549, 0.3521, 0.1412, 0.1412];
-        set bondCoefficients[47] = [0.1078, 0.0572, 0.3491, 0.1422, 0.1422];
-        set bondCoefficients[48] = [0.1059, 0.0594, 0.3461, 0.1432, 0.1432];
-        set bondCoefficients[49] = [0.1042, 0.0614, 0.3433, 0.1441, 0.1441];
-        set bondCoefficients[50] = [0.1026, 0.0632, 0.3406, 0.145, 0.145];
-        set bondCoefficients[51] = [0.1011, 0.0649, 0.3379, 0.1458, 0.1458];
-        set bondCoefficients[52] = [0.0997, 0.0665, 0.3354, 0.1467, 0.1467];
-        set bondCoefficients[53] = [0.0984, 0.0679, 0.3329, 0.1475, 0.1475];
+        let bondCoefficients = new Double[][nBondLengths]
```
Comment (bettinaheim) [inline Samples/src/H2SimulationGUI/H2SimulationOperations.qs]: same here

File: Samples/tests/SampleTests/BayesianPETests.qs [hunk 4]
```diff
@@ -5,13 +5,13 @@ namespace Microsoft.Quantum.Tests {
     open Microsoft.Quantum.Canon;
     open Microsoft.Quantum.Diagnostics;
 
-    operation BayesianPEManualTest() : Unit {
+    // operation BayesianPEManualTest() : Unit {
```
Comment (bettinaheim) [inline Samples/tests/SampleTests/BayesianPETests.qs]: Not sure if you meant to commit this. I assume you may want to remove this test temporarily until I update the compiler?

File: Samples/tests/SampleTests/SimpleAlgorithmsTests.qs [hunk 5]
```diff
@@ -73,27 +74,24 @@ namespace Microsoft.Quantum.Tests {
         
         // setting up a few test cases of both, constant and balanced functions.
         let testList = [
-            (2, [1, 2], false), 
+            // (2, [1, 2], false),
```
Comment (bettinaheim) [inline Samples/tests/SampleTests/SimpleAlgorithmsTests.qs]: why is this commented out?

File: Chemistry/SimulateHubbardHamiltonian/HubbardSimulation.qs (hunk 1 above)
Comment (cgranade) [reply Chemistry/SimulateHubbardHamiltonian/HubbardSimulation.qs]: Definitely agreed. My goal with this PR was to convert things literally, but I think this snippet could be refactored to use an array function pretty readily.

File: Samples/src/H2SimulationCmdLine/Operation.qs (hunk 2 above)
Comment (cgranade) [reply Samples/src/H2SimulationCmdLine/Operation.qs]: Agreed, `mutable` isn't needed here. My understanding was that this design was meant to emphasize the indices for each element, but I'll convert to a literal.

File: Samples/src/H2SimulationGUI/H2SimulationOperations.qs (hunk 3 above)
Comment (cgranade) [reply Samples/src/H2SimulationGUI/H2SimulationOperations.qs]: Agreed.

File: Samples/tests/SampleTests/BayesianPETests.qs (hunk 4 above)
Comment (cgranade) [reply Samples/tests/SampleTests/BayesianPETests.qs]: I did not mean to comment this, it was a temporary debugging thing I forgot to uncomment again. Thanks for catching!

File: Samples/tests/SampleTests/SimpleAlgorithmsTests.qs (hunk 5 above)
Comment (cgranade) [reply Samples/tests/SampleTests/SimpleAlgorithmsTests.qs]: Same as above.

File: Chemistry/SimulateHubbardHamiltonian/HubbardSimulation.qs (hunk 1 above)
Comment (cgranade) [reply Chemistry/SimulateHubbardHamiltonian/HubbardSimulation.qs]: Looking again, that array isn't used anywhere? I've asked @guanghaolow and will update or fix accordingly. Thanks for catching this as well!

Comment (cgranade) [reply Chemistry/SimulateHubbardHamiltonian/HubbardSimulation.qs]: After discussing with @guanghaolow, I believe this was a minor bug, and have fixed it accordingly, and have added him as a reviewer for the PR. Thanks again for catching this, @bettinaheim!

**Pull request merged**